Sources/TSCBasic/FileSystem.swift

```diff
@@ -558,7 +571,16 @@ public class InMemoryFileSystem: FileSystem {
             }
 
             // Return the directory entry.
-            return contents.entries[path.basename]
+            let node = contents.entries[path.basename]
+
+            switch node?.contents {
+            case .directory, .file:
+                return node
+            case .symlink(let destination):
+                return followSymlink ? try getNodeInternal(destination) : node
```

Comment:
If the payload of the symlink is a string, this would then be something like `AbsolutePath(destination, relativeTo: path.parentDirectory)`.